app/Models/Beatmapset.php

```diff
@@ -1314,6 +1221,45 @@ public function hasFullBNNomination($mode = null)
             });
     }
 
+    public function nominationsByType()
+    {
+        $nominations = $this->beatmapsetNominations()
+            ->current()
+            ->with('user')
+            ->get();
+
+        $result = [
+            'full' => [],
+            'limited' => [],
+        ];
+
+        foreach ($nominations as $nomination) {
+            $userNominationModes = $nomination->user->nominationModes();
+            // no permission
+            if ($userNominationModes === null) {
+                continue;
+            }
+
+            // legacy nomination, only check group
+            if ($nomination->modes === null) {
+                if ($nomination->user->isLimitedBN()) {
+                    $result['limited'][] = null;
+                } else if ($nomination->user->isBNG() || $nomination->user->isNAT()) {
+                    $result['full'][] = null;
+                }
+            } else {
+                foreach ($nomination->modes as $mode) {
+                    $nominationType = $userNominationModes[$mode] ?? null;
```

Comment:
this would potentially invalidate the nominations displayed on web, wouldn't it?